### PR TITLE
returned the missing files

### PR DESCRIPTION
**Pull request by memmas**

### CODE REVIEW

File: app/src/main/java/org/ieeeguc/ieeeguc/controllers/LoginActivity.java [hunk 1]
```diff
@@ -1,28 +1,210 @@
...
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import static org.ieeeguc.ieeeguc.models.User.login;
 
 /**
  * A login screen that offers login via email/password.
  */
 public class LoginActivity extends AppCompatActivity {
 
+    AutoCompleteTextView email ;
```
Comment (KEMDAK) [inline app/src/main/java/org/ieeeguc/ieeeguc/controllers/LoginActivity.java]: change it to EditText

File: app/src/main/java/org/ieeeguc/ieeeguc/controllers/LoginActivity.java [hunk 2]
```diff
@@ -1,28 +1,210 @@
...
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_login);
-    }
+        email = (AutoCompleteTextView) findViewById(R.id.email) ;
+        password = (EditText) findViewById(R.id.password) ;
+        send = (Button) findViewById(R.id.email_sign_in_button) ;
+        final String[] message = new String[1];
+        send.setOnClickListener(new OnClickListener() {
+            @Override
+            public void onClick(View view) {
```
Comment (KEMDAK) [inline app/src/main/java/org/ieeeguc/ieeeguc/controllers/LoginActivity.java]: move the code of this method to an external method that is called `attemptLogin();` and call it here

File: app/src/main/java/org/ieeeguc/ieeeguc/controllers/LoginActivity.java [hunk 3]
```diff
@@ -1,28 +1,210 @@
...
+                String emailText = email.getText().toString() ;
+                String passwordText = password.getText().toString() ;
+
+
+                if (emailText.length()!= 0 && passwordText.length()!= 0 ) {
+                    login(emailText, passwordText, new HTTPResponse() {
+
+                        @Override
+                        public void onSuccess(int statusCode, JSONObject body)  {
+                            Log.i("2","it is clicked");
```
Comment (KEMDAK) [inline app/src/main/java/org/ieeeguc/ieeeguc/controllers/LoginActivity.java]: remove the testing logs all of them

File: app/src/main/java/org/ieeeguc/ieeeguc/controllers/LoginActivity.java [hunk 4]
```diff
@@ -1,28 +1,210 @@
...
+                                    SharedPreferences Sp = getApplicationContext().getSharedPreferences(getString(R.string.shared_preferences_name), Context.MODE_PRIVATE);
+                                    SharedPreferences.Editor E = Sp.edit();
+                                    E.putString("token", token);
+                                    E.commit();
+
+                                    MainActivity.token = token ;
+
+
+
+                                    JSONObject USer = body.getJSONObject("user") ;
```
Comment (KEMDAK) [inline app/src/main/java/org/ieeeguc/ieeeguc/controllers/LoginActivity.java]: rename the variable according to the conventions

File: app/src/main/java/org/ieeeguc/ieeeguc/controllers/LoginActivity.java [hunk 5]
```diff
@@ -1,28 +1,210 @@
...
+                                    int id = USer.getInt("id") ;
+                                    String TYPE = USer.getString("type") ;
```
Comment (KEMDAK) [inline app/src/main/java/org/ieeeguc/ieeeguc/controllers/LoginActivity.java]: rename the variable according to the conventions

File: app/src/main/java/org/ieeeguc/ieeeguc/controllers/LoginActivity.java [hunk 6]
```diff
@@ -1,28 +1,210 @@
...
+                                    SimpleDateFormat dateFormatter = new SimpleDateFormat("yyyy-mm-dd");
+                                    Date BD = dateFormatter.parse(BDS.substring(0,10));
+                                    String IEEE_membership_ID = USer.getString("IEEE_membership_ID") ;
+                                    JSONObject settings = USer.getJSONObject("settings") ;
+
+                                    if(USer.has("committee")  ){
+                                        JSONObject committee =USer.getJSONObject("committee") ;
+                                        String committeeName =committee.getString("committeeName") ;
+                                        int committeeID = committee.getInt("committeeID") ;
+                                        User user = new User(id,type,FN,LN,GENDER,email,BD ,IEEE_membership_ID,committeeID,committeeName,PN,settings) ;
```
Comment (KEMDAK) [inline app/src/main/java/org/ieeeguc/ieeeguc/controllers/LoginActivity.java]: take the last four lines common in the then and the else part

File: app/src/main/java/org/ieeeguc/ieeeguc/controllers/LoginActivity.java [hunk 7]
```diff
@@ -1,28 +1,210 @@
...
+                                @Override
+                                public void onClick(View view) {
+
+                                }
+                            })
+                                    .show();
+                            Log.i("2","your email is null");
+                        }
+
+                        else{
```
Comment (KEMDAK) [inline app/src/main/java/org/ieeeguc/ieeeguc/controllers/LoginActivity.java]: this else will never be executed trace your code

File: app/src/main/java/org/ieeeguc/ieeeguc/models/User.java [hunk 8]
```diff
@@ -95,6 +98,64 @@ public String getCommitteeName() {
         return committeeName;
     }
 
+    public static void ForgetPassword(String email, final HTTPResponse HTTPR) {
```
Comment (KEMDAK) [inline app/src/main/java/org/ieeeguc/ieeeguc/models/User.java]: add documentation to your method look at @rehamayman96 code.
Follow the naming conventions of variables

File: app/src/main/java/org/ieeeguc/ieeeguc/models/User.java [hunk 9]
```diff
@@ -95,6 +98,64 @@ public String getCommitteeName() {
...
+            public void onFailure(Call call, IOException e) {
+                e.printStackTrace();
+            }
+
+            @Override
+            public void onResponse(Call call, final Response response) {
+                try {
+                    HTTPR.onSuccess(200,new JSONObject(response.body().toString())) ;
+                } catch (JSONException e) {
+                    e.printStackTrace();
```
Comment (KEMDAK) [inline app/src/main/java/org/ieeeguc/ieeeguc/models/User.java]: call `HTTPResponse.onFailure()` if the parsing failed

File: app/src/main/java/org/ieeeguc/ieeeguc/models/User.java [hunk 10]
```diff
@@ -95,6 +98,64 @@ public String getCommitteeName() {
...
+                }
```
Comment (KEMDAK) [inline app/src/main/java/org/ieeeguc/ieeeguc/models/User.java]: add `response.close()` at the end of the `onResponse`

File: app/src/main/java/org/ieeeguc/ieeeguc/models/User.java [hunk 11]
```diff
@@ -95,6 +98,64 @@ public String getCommitteeName() {
...
+        HashMap <String,String>p = new HashMap();
+        p.put("email",email) ;
+        OkHttpClient client = new OkHttpClient();
+        final Request request = new Request.Builder()
+                .url("http://ieeeguc.org/api/login")
+                .post(RequestBody.create(contentType,( new JSONObject(p)).toString()))
+                .build();
+        client.newCall(request).enqueue(new Callback() {
+            public void onFailure(Call call, IOException e) {
+                e.printStackTrace();
```
Comment (KEMDAK) [inline app/src/main/java/org/ieeeguc/ieeeguc/models/User.java]: call `HTTPResponse.onFailure()`. and add `call.cancel()`

File: app/src/main/java/org/ieeeguc/ieeeguc/models/User.java [hunk 12]
```diff
@@ -95,6 +98,64 @@ public String getCommitteeName() {
...
+                try {
+                    HTTPR.onSuccess(200,new JSONObject(response.body().toString())) ;
+                } catch (JSONException e) {
+                    e.printStackTrace();
+                }
+            }
+        });
+    }
+
+    public static void login(String email , String password ,final HTTPResponse HTTP_RESPONSE){
```
Comment (KEMDAK) [inline app/src/main/java/org/ieeeguc/ieeeguc/models/User.java]: add documentation to your code and follow the naming conventions of variables

File: app/src/main/java/org/ieeeguc/ieeeguc/models/User.java [hunk 13]
```diff
@@ -95,6 +98,64 @@ public String getCommitteeName() {
...
+        final JSONObject jsonB = new JSONObject();
+        try{
+            jsonB.put("email",email);
+            jsonB.put("password",password);
+        }catch(JSONException e){
+            e.printStackTrace();
```
Comment (KEMDAK) [inline app/src/main/java/org/ieeeguc/ieeeguc/models/User.java]: you have to call `HTTPResponse.onFailure()` when your method fails.
Also you can't continue execution when the json object that you are trying to send is failed to be constructed

File: app/src/main/java/org/ieeeguc/ieeeguc/models/User.java [hunk 14]
```diff
@@ -95,6 +98,64 @@ public String getCommitteeName() {
...
+        OkHttpClient client = new OkHttpClient();
+        RequestBody body = RequestBody.create(contentType, jsonB.toString());
+        Request request = new Request.Builder()
+                .url("http://ieeeguc.org/api/login")
+                .header("user_agent","Android")
+                .post(body)
+                .build();
+        client.newCall(request).enqueue(new Callback() {
+            public void onFailure(Call call, IOException e) {
+                e.printStackTrace();
```
Comment (KEMDAK) [inline app/src/main/java/org/ieeeguc/ieeeguc/models/User.java]: call `HTTPResponse.onFailure()`. and add `call.cancel()`

File: app/src/main/java/org/ieeeguc/ieeeguc/models/User.java [hunk 15]
```diff
@@ -95,6 +98,64 @@ public String getCommitteeName() {
...
+                    int x = response.code();
+                    String y = Integer.toString(x);
+                    if(y.charAt(0)== '2'){
+                        HTTP_RESPONSE.onSuccess(response.code(),json);
+                    }
+                    else{
+                        HTTP_RESPONSE.onFailure(response.code(),json);
+                    }
+                } catch (JSONException e) {
+                    HTTP_RESPONSE.onFailure(response.code(),null);}}});}
```
Comment (KEMDAK) [inline app/src/main/java/org/ieeeguc/ieeeguc/models/User.java]: indent your code...!!
and send the failure code 500.
And add the `response.close()` at the end of `onResponse`

File: app/src/main/res/values/strings.xml [hunk 16]
```diff
@@ -13,11 +13,19 @@
     <string name="error_connection">Operation Failed, please check your internet Connection!</string>
     <string name="error_server_down">Sorry something went wrong with the server,please try again later!</string>
 
+    <!--Strings related to SharedPreferences-->
+    <string name="shared_preferences_name">preferences_ieee_guc</string>
+    <string name ="error_401">your email or password is wrong </string>
```
Comment (KEMDAK) [inline app/src/main/res/values/strings.xml]: all the strings bellow are redundant

Comment (KEMDAK) [reply app/src/main/res/values/strings.xml]: and they don't belong to this section of the file

**Review by KEMDAK — CHANGES_REQUESTED** [review]

File: app/src/main/java/org/ieeeguc/ieeeguc/controllers/LoginActivity.java (hunk 7 above)
Comment (memmas) [reply app/src/main/java/org/ieeeguc/ieeeguc/controllers/LoginActivity.java]: it is working i tried it on the app

**Review by KEMDAK — APPROVED** [review]

**Pull request merged**